### PR TITLE
Move static folder

### LINKED ISSUE #467: Move `static` inside `docsrc`
Very minor detail, but to avoid clutter, we could move the static images inside docsrc. Would require updating all the relative image paths - I'm happy to do it if you agree @till-m?

### PR DESCRIPTION
**Pull request by bwheelz36**
this turned out to be a lot more work/ changes than I expected...

- Fixes #467 
- Updates the build_docs CI job to use pyproject.toml and poetry
- also added a link to the scikit learn docs on different GP kernels to advanced tour notebook

### CODE REVIEW

File: .github/workflows/build_docs.yml [hunk 1]
```diff
@@ -22,21 +22,15 @@ jobs:
...
-          pip install myst-parser
-      - name: Install package
-        run: |
-          pip install -e .
+          poetry install --with dev,nbtools
       - name: build sphinx docs
         run: |
           cd docsrc
-          make github
+          poetry run make github
```
Comment (till-m) [inline .github/workflows/build_docs.yml:33]: Is there a reason why we need to run these commands with poetry?

File: docsrc/Makefile [hunk 2]
```diff
@@ -15,10 +15,10 @@ BUILDDIR      = ../docs
 .PHONY: help Makefile
 
 github:
-	@cp ../README.md .
-	@cp -r ../static .
+# 	@cp ../README.md .
 	@make html
 	@cp -a ../docs/html/. ../docs
+	@ cp -r ../docsrc/ ../docs
```
Comment (till-m) [inline docsrc/Makefile:21]: Nitpick

```suggestion
	@cp -r ../docsrc/ ../docs
```

**Review by till-m — APPROVED** [review]

File: .github/workflows/build_docs.yml (hunk 1 above)
Comment (bwheelz36) [reply .github/workflows/build_docs.yml:33]: No, just the easiest way to activate the venv. You don't like it?

Comment (till-m) [reply .github/workflows/build_docs.yml:33]: Ah I see otherwise the job fails. All good, thanks

**Pull request merged**